mifosng-provider/src/main/java/org/mifosplatform/infrastructure/documentmanagement/api/ImagesApiResource.java

```diff
@@ -70,9 +61,9 @@ public ImagesApiResource(final PlatformSecurityContext context, final ImageReadP
     @POST
     @Consumes({ MediaType.MULTIPART_FORM_DATA })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String addNewClientImage(@PathParam("clientId") final Long clientId, @HeaderParam("Content-Length") final Long fileSize,
-            @FormDataParam("file") final InputStream inputStream, @FormDataParam("file") final FormDataContentDisposition fileDetails,
-            @FormDataParam("file") final FormDataBodyPart bodyPart) {
+    public String addNewClientImage(@PathParam("clients") final String clientName, @PathParam("clientId") final Long clientId,
```

Comment:
I don't think this applies here. In this case "clientName" is taken directly from the HTTP request URI and defines what type of object to process. It can be either "clients" or "staff" which are mapped to different objects and the id is related to the object type (staff.id!=client.id)